app/kumactl/cmd/install/install_transparent_proxy.go

```diff
@@ -0,0 +1,192 @@
+    - if using systemd to run add 'User=kuma-dp' in the '[Service]' section of the service file
+    - leverage 'runuser' similar to (assuming aforementioned yaml):
+
+runuser -u kuma-dp -- \
+  /usr/bin/kuma-dp run \
+    --cp-address=https://172.19.0.2:5678 \
+    --dataplane-token-file=/kuma/token-demo \
+    --dataplane-file=/kuma/dpyaml-demo \
+    --dataplane-var name=dp-demo \
+    --dataplane-var address=172.19.0.4 \
+    --dataplane-var port=80  \
+    --binary-path /usr/local/bin/envoy
+
+`,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			if !args.DryRun && runtime.GOOS != "linux" {
+				return errors.Errorf("transparent proxy will work only on Linux OSes")
+			}
+
+			if args.ModifyIptables && args.User == "" && args.UID == "" {
+				return errors.Errorf("--kuma-dp-user or --kuma-dp-uid should be supplied")
+			}
+
+			if args.ModifyResolvConf && args.KumaCpIP.String() == net.IPv4(0, 0, 0, 0).String() {
+				return errors.Errorf("please supply a valid `--kuma-cp-ip`")
+			}
+
+			if args.ModifyIptables {
+				tp := transparentproxy.GetDefaultTransparentProxy()
+
+				uid, gid, err := findUidGid(args.UID, args.User)
+				if err != nil {
+					return errors.Wrapf(err, "unable to find the kuma-dp user")
+				}
+
+				output, err := tp.Setup(&config.TransparentProxyConfig{
+					DryRun:               args.DryRun,
+					RedirectPortOutBound: args.RedirectPortOutBound,
+					RedirectPortInBound:  args.RedirectPortInBound,
+					ExcludeInboundPorts:  args.ExcludeInboundPorts,
+					ExcludeOutboundPorts: args.ExcludeOutboundPorts,
+					UID:                  uid,
+					GID:                  gid,
+				})
+				if err != nil {
+					return errors.Wrap(err, "failed to setup transparent proxy")
+				}
+
+				if args.DryRun {
+					_, _ = cmd.OutOrStdout().Write([]byte(output))
+				}
+			}
+
+			if args.ModifyResolvConf {
+				kumaCPLine := fmt.Sprintf("nameserver %s", args.KumaCpIP.String())
+				content, err := ioutil.ReadFile("/etc/resolv.conf")
+				if err != nil {
+					return errors.Wrap(err, "unable to open /etc/resolv.conf")
+				}
+				newcontent := fmt.Sprintf("%s\n%s", kumaCPLine, string(content))
+
+				if !args.DryRun && !strings.Contains(string(content), kumaCPLine) {
+					err = ioutil.WriteFile("/etc/resolv.conf.kuma", content, 0644)
+					if err != nil {
+						return errors.Wrap(err, "unable to open /etc/resolv.conf.kuma")
+					}
+					err = ioutil.WriteFile("/etc/resolv.conf", []byte(newcontent), 0644)
+					if err != nil {
+						return errors.Wrap(err, "unable to write /etc/resolv.conf")
+					}
+				}
+
+				if args.DryRun {
+					_, _ = cmd.OutOrStdout().Write([]byte(newcontent))
+					_, _ = cmd.OutOrStdout().Write([]byte("\n"))
+				} else {
+					content, err := ioutil.ReadFile("/etc/resolv.conf")
+					if err != nil {
+						return errors.Wrap(err, "uanble to open /etc/resolv.conf")
+					}
+					_, _ = cmd.OutOrStdout().Write(content)
+					_, _ = cmd.OutOrStdout().Write([]byte("\n"))
```

Comment:
All applied except for modifying the file, as it will make restoring it slightly harder. But meanwhile, its name is still saying enough (kuma-backup).